src/clj/puppetlabs/http/client/async.clj

```diff
@@ -242,7 +152,7 @@
    * :query-params - used to set the query parameters of an http request"
   [opts :- common/RawUserRequestOptions
    callback :- common/ResponseCallbackFn
-   client :- HttpAsyncClient]
+   client :- common/Client]
```

Comment:
I guess I would rather see us keep `common/Client`, in case we need it in other places later (it was already being used at https://github.com/rlinehan/clj-http-client/blob/client-init-refactor/src/clj/puppetlabs/http/client/async.clj#L25), but maybe we can just remove that TODO? It seems like it is really old, and we now have the `common/HTTPClient` protocol now.